clientapi/routing/membership.go

```diff
@@ -388,15 +388,20 @@ func checkMemberInRoom(ctx context.Context, stateAPI currentstateAPI.CurrentStat
 		e := jsonerror.InternalServerError()
 		return &e
 	}
-	ev, ok := membershipRes.StateEvents[tuple]
-	if !ok {
+	ev := membershipRes.StateEvents[tuple]
+	if ev == nil {
 		return &util.JSONResponse{
 			Code: http.StatusForbidden,
 			JSON: jsonerror.Forbidden("user does not belong to room"),
 		}
 	}
 	membership, err := ev.Membership()
-	if err != nil || membership != "join" {
+	if err != nil {
+		util.GetLogger(ctx).WithError(err).Error("Member event isn't valid")
```

Comment:
More accurately, `Membership()` returns an error when it's *either* an invalid membership state *or* not a membership event.

Reply:
Yes, but since we requested events of the type `gomatrixserverlib.MRoomMember` and of the current user, this should only error on wrong content.